### ISSUE TITLE
delay in loader display

### ISSUE THREAD

**Issue by rohankm** [tomcru/holy-loader#22]
hi im using `router.push()` from next/navigation. whenever I try to navigate the loader doesn't start on the go.. at the very end of the page load it starts showing.

**Comment by tomcru** [comment]
This is due to `router.push()` only firing at the end (when your page has loaded). I'm not aware of any other events to listen to - to trigger the loader earlier. Do you have any suggestions?

**Comment by rohankm** [comment]
Is it better to create a new router hook where we can manually track when the router.push is run

**Comment by monkeystylle** [comment]
hi is there an update or fix for this? the delay when using router.push ?

**Comment by tomcru** [comment]
I don't believe this is possible, because the router.push event only fires once the page has already loaded. Unless you are aware of another way we could implement this.

We did, however, just add 2 new funcs with which you can manually start & stop the loader:

```typescript
'use client';

import { startHolyLoader, stopHolyLoader } from 'holy-loader';

startHolyLoader();
stopHolyLoader();
```

So you should be able to do:
```typescript
startHolyLoader();
router.push('/your-page')
```

**Comment by monkeystylle** [comment]
hey thanks, after reading the docs, using startHolyLoader(); before router.push works.. thanks a lot

**Comment by seppemaqet** [comment]
I've been experimenting with using the [onRouterTransitionStart](https://nextjs.org/docs/app/api-reference/file-conventions/instrumentation-client#router-navigation-tracking), which does seem to fire synchronously after calling `router.push()`. I've currently implemented it like this:

```
// instrumentation.client.ts

import { startHolyLoader, stopHolyLoader } from 'holy-loader';

export function onRouterTransitionStart(href: string, navigationType: string) {
  if (navigationType === 'push') {
    startHolyLoader();
  }
}
```

This seems to work well enough in the cases that I tested. Although I think I should replicate the `isSamePathname` etc. checks that are done in [`stopProgressHistoryOnUpdate`](https://github.com/tomcru/holy-loader/blob/b4c1a7d05080fe0cd7863890ad915e1fc4c7a650/src/index.tsx#L151) (and even then there may be cases where the loader doesn't get cancelled properly.)

I'm not sure this is a robust solution, but I wanted to leave this here in case you weren't aware this hook exists.
